sources/common/infoschema.go

```diff
@@ -63,10 +63,17 @@ type FkConstraint struct {
 // and we use them to obtain source database's schema information.
 func ProcessSchema(conv *internal.Conv, infoSchema InfoSchema, numWorkers int, attributes internal.AdditionalSchemaAttributes) error {
 
-	GenerateSrcSchema(conv, infoSchema, numWorkers)
+	tableCount, err := GenerateSrcSchema(conv, infoSchema, numWorkers)
+	if err != nil {
+		return err
+	}
 	initPrimaryKeyOrder(conv)
 	initIndexOrder(conv)
 	SchemaToSpannerDDL(conv, infoSchema.GetToDdl())
+	if tableCount != len(conv.SpSchema) {
+		fmt.Printf("Failed to load all the source tables, source table count: %v, processed tables:%v\n", tableCount, len(conv.SpSchema))
+		return fmt.Errorf("failed to load all the source tables, source table count: %v, processed tables:%v", tableCount, len(conv.SpSchema))
```

Comment:
We should also add the action expected from the user on seeing this error. 
Example - Please retry connecting to the source database to load tables.